src/hooks/useSwapper/useSwapper.ts

```diff
@@ -0,0 +1,184 @@
+import { SwapperManager, ZrxSwapper } from '@shapeshiftoss/swapper'
+import { Asset, GetQuoteInput, Quote, SwapperType } from '@shapeshiftoss/types'
+import { debounce } from 'lodash'
+import { useCallback, useState } from 'react'
+import { useFormContext, useWatch } from 'react-hook-form'
+import { TradeAsset, TradeState } from 'components/Trade/Trade'
+import { useChainAdapters } from 'context/ChainAdaptersProvider/ChainAdaptersProvider'
+import { bn } from 'lib/bignumber/bignumber'
+import { fromBaseUnit, toBaseUnit } from 'lib/math'
+import { web3Instance } from 'lib/web3-instance'
+
+const debounceTime = 1000
+
+export enum TradeActions {
+  BUY = 'BUY',
+  SELL = 'SELL',
+  FIAT = 'FIAT'
+}
+
+type GetQuote = {
+  amount: Pick<GetQuoteInput, 'buyAmount' | 'sellAmount'>
+  sellAsset: Asset
+  buyAsset: Asset
+  onFinish: (quote: Quote) => void
+  action?: TradeActions
+}
+
+export const useSwapper = () => {
```

Comment:
There is a lot going on in this hook and I feel like it could benefit from some tests

Reply:
Look at my comment above. That is the next ticket I am picking up